core/src/main/java/com/linecorp/armeria/client/endpoint/StickyEndpointSelectionStrategy.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.function.ToLongFunction;
+
+import com.google.common.hash.Hashing;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * An {@link EndpointSelector} strategy which implements sticky load-balancing using
+ * user passed {@link ToLongFunction<ClientRequestContext>} as hash calculator for
```

Comment:
`using a user passed... to compute hashes for consistent hashing.`

Also, might be user-friendly to add an example, e.g.

```
<p>This can be useful when all requests that qualify some given criteria must be sent to the same backend server. A common use case is to send all requests for the same logged-in user to the same backend, which could have a local cache keyed by user id.
```